trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/generate_propeller_wake_distribution.py

```diff
@@ -0,0 +1,209 @@
+## @ingroup Methods-Aerodynamics-Common-Fidelity_Zero-Lift
+#  generate_propeller_wake_distribution.py
+# 
+# Created:  Jul 2020, M. Clarke 
+
+# ----------------------------------------------------------------------
+#  Imports
+# ----------------------------------------------------------------------
+
+# package imports
+import numpy as np
+from SUAVE.Core import Data 
+from SUAVE.Methods.Aerodynamics.Common.Fidelity_Zero.Lift.compute_wake_contraction_matrix import compute_wake_contraction_matrix
+## @ingroup Methods-Aerodynamics-Common-Fidelity_Zero-Lift  
+
+def generate_propeller_wake_distribution(prop,m,VD,init_timestep_offset):
+    # to put in settings 
+    time = 0.05
+    
+    # Unpack unknowns  
+    R            = prop.tip_radius
+    r            = prop.radius_distribution 
+    c            = prop.chord_distribution 
+    Na           = prop.outputs.number_azimuthal_stations
+    Nr           = prop.outputs.number_radial_stations
+    omega        = prop.outputs.omega                        
+    vt           = prop.outputs.disc_tangential_induced_velocity         
+    va           = prop.outputs.disc_axial_induced_velocity 
+    V_inf        = prop.outputs.velocity
+    MCA          = prop.mid_chord_aligment 
+    B            = prop.number_blades
+    gamma        = prop.outputs.disc_circulation
+    blade_angles = np.linspace(0,2*np.pi,B+1)[:-1]        
+    dt           = (2*np.pi/Na)/omega[0]
+    nts          = int(time/dt)
+    ts           = np.linspace(0,time,nts)
+    num_prop     = len(prop.origin) 
+    
+    t0           = dt*init_timestep_offset
+    start_angle  = omega[0]*t0
+    blade_angles = blade_angles  
+    
+    # define points ( control point, time step , blade number , location on blade )
+    # compute lambda and mu 
+    mean_induced_velocity  = np.mean( np.mean(va,axis = 1),axis = 1)  
+    
+    lambda_tot   =  np.atleast_2d((V_inf[:,0]  + mean_induced_velocity)).T /(omega*R)       # inflow advance ratio (page 30 Leishman)
+    mu_prop      =  np.atleast_2d(V_inf[:,2]).T /(omega*R)                              # rotor advance ratio  (page 30 Leishman) 
+    V_prop       =  np.atleast_2d(np.sqrt((V_inf[:,0]  + mean_induced_velocity)**2 + (V_inf[:,2])**2)).T
+    
+    # wake skew angle 
+    wake_skew_angle = np.arctan(mu_prop/lambda_tot)
+    
+    # reshape gamma to find the average between stations 
+    gamma_new = np.zeros((m,Na,(Nr-1)))
+    gamma_new = (gamma[:,:,:-1] + gamma[:,:,1:])*0.5
+    
+    # define empty arrays 
+    WD       = Data()
+    n        = Nr-1
+    WD_XA1   = np.zeros((m,num_prop,nts-1,B,n))
```

Comment:
you should preallocate all of these sizes once as tuples

Reply:
also you could do more compact preallocation with a for loop, a map, or a generator

Reply:
The size of these matrices will change 1) if you have a combination of propellers/rotors with different blades, the number of each type of rotor/propeller etc. Here is therefore the best place to put them. I did not want to store them anywhere else because there is no need for creating empty matrices/vectors if I am not going to use a distributed vortex ring wake